app/GioForegroundService.java

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: Unlicense OR MIT
+
+package org.gioui;
+import android.app.Notification;
+import android.app.Service;
+import android.app.Notification;
+import android.app.Notification.Builder;
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.os.IBinder;
+
+public class GioForegroundService extends Service {
+	public static final String CHANNEL_ID = "GioForegroundService";
+	public static final String serviceName = "GioForeground";
+	public static final int notificationID = 42;
+
+	@Override public int onStartCommand (Intent intent, int flags, int startId) {
+		return START_NOT_STICKY;
+	}
+	@Override public IBinder onBind(Intent intent) {
+		return null;
+	}
+	@Override public void onCreate() {
+		super.onCreate();
+		this.createNotificationChannel();
+		this.startForeground();
+	}
+	public void startForeground() {
+		// create the Intent that will bring GioActivity to foreground
+		Context ctx = getApplicationContext();
+		Intent resultIntent = new Intent(ctx, GioActivity.class);
+		PendingIntent pending = PendingIntent.getActivity(ctx, notificationID, resultIntent, Intent.FLAG_ACTIVITY_CLEAR_TASK);
+		// create the Notification that Android will display on the tray for this service
+		Notification.Builder builder = new Notification.Builder(ctx, CHANNEL_ID)
+			.setContentTitle(CHANNEL_ID)
+			.setSmallIcon(getResources().getIdentifier("@mipmap/ic_launcher", "drawable", getPackageName()))
+			.setContentText(serviceName)
+			.setContentIntent(pending)
+			.setPriority(Notification.PRIORITY_DEFAULT);
```

Comment:
Lower priority? See also comment below about importance.

Reply:
I'm open to changing this but I think it would be more useful if these sort of choices were available to the application developer - although unfortunately this is rather platform specific. I don't think android wants you to be able to hide these notifications from the user in any case, so I'm not really sure what adjusting the priority does - but I will try to find out.

Reply:
Not hide, so much as to give the notification a lower visibility. For example, I believe lower priority notifications do not appear when you drag down the system bar to reveal notification details.